tests/virt_autotest/virsh_internal_snapshot.pm

```diff
@@ -35,6 +35,13 @@ sub run_test {
     foreach my $guest (keys %xen::guests) {
         my $type = check_guest_disk_type($guest);
         next if ($type == 1);
+        record_info "virsh-snapshot", "Cleaning in case of rerun";
+        if (script_run("virsh snapshot-list $guest | grep internal-snapshot-$guest-01") != 0) {
+            assert_script_run "virsh snapshot-delete $guest --snapshotname internal-snapshot-$guest-01 || true";
+        }
+        if (script_run("virsh snapshot-list $guest | grep internal-snapshot-$guest-02") != 0) {
+            assert_script_run "virsh snapshot-delete $guest --snapshotname internal-snapshot-$guest-02 || true";
```

Comment:
I would recommend "assert_script_run "virsh snapshot-delete $guest --snapshotname internal-snapshot-$guest-02", although it is not check point, but it is necessary preparation step(if it fails, no further test can be done). This is usually a strategy of us to necessary preparation steps.

But if you do not agree, it is fine. In this case, I would recommend to directly use script_run for simplicity.

Reply:
Similar comment for snapshot-01.

Reply:
All others LGTM.